src/workers/block_listenser.rs

```diff
@@ -284,15 +284,44 @@ impl BlockListener {
         postgres: Option<PostgresMpscSend>,
     ) -> JoinHandle<anyhow::Result<()>> {
         tokio::spawn(async move {
+            let commitment = commitment_config.commitment;
+
+            info!("Listening to {commitment:?} blocks");
+
             loop {
-                let ws_addr = &self.tx_sender.tpu_manager.ws_addr;
-                let pub_sub_client = PubsubClient::new(ws_addr).await?;
-                let err = self
-                    .clone()
-                    .listen_from_pubsub(&pub_sub_client, commitment_config, &postgres)
-                    .await
-                    .unwrap_err();
-                warn!("{commitment_config:?} Block Subscribe error {err}");
+                let (
+                    _,
+                    BlockInformation {
+                        slot: latest_slot,
+                        block_height: _,
+                    },
+                ) = self
+                    .block_store
+                    .get_latest_block_info(commitment_config)
+                    .await;
+
+                let block_slots = self
+                    .rpc_client
+                    .get_blocks_with_commitment(latest_slot, None, commitment_config)
+                    .await?;
+
+                let block_future_handlers = block_slots.into_iter().map(|slot| {
```

Comment:
this will spawn too many requests in parallel and overload the rpc. potentially causing missing slots in your block_store